### PR TITLE
AQI: Added API key provider

### PR DESCRIPTION
**Pull request by Boernsman**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

- [x] If you added a new plugin, should it be added to nymea-plugins-all or nymea-plugins-maker?

### CODE REVIEW

File: aqi/integrationpluginaqi.cpp [hunk 1]
```diff
@@ -30,63 +30,27 @@
...
-    connect(reply, &QNetworkReply::finished, this, [reply, info] {
-        reply->deleteLater();
-
-        if (reply->error() == QNetworkReply::NetworkError::HostNotFoundError) {
-            info->finish(Thing::ThingErrorHardwareNotAvailable, QT_TR_NOOP("Air quality index server is not reachable."));
-        } else {
-            info->finish(Thing::ThingErrorNoError, QT_TR_NOOP("Please enter your API token for Air Quality Index"));
+        if (paramTypeId == airQualityIndexPluginApiKeyParamTypeId && m_aqiConnection) {
+            if (!value.toString().isEmpty())
+                m_aqiConnection->setApiKey(value.toString());
```
Comment (mzanetti) [inline aqi/integrationpluginaqi.cpp:43]: uhmm... Wouldn't this crash if you set the API without actually having a AQI thing set up?

File: aqi/integrationpluginaqi.json [hunk 2]
```diff
@@ -1,11 +1,21 @@
 {
     "name": "AirQualityIndex",
-    "displayName": "AirQualityIndex",
+    "displayName": "Air quality index",
     "id": "57d69b76-4d2d-41ec-bef6-949a79ffbe6b",
+    "paramTypes": [
+        {
+            "id": "b6861adb-7ed5-445f-b500-4df9eab866ef",
+            "name": "apiKey",
+            "displayName": "API key",
```
Comment (mzanetti) [inline aqi/integrationpluginaqi.json:9]: "Custom API key" would IMO be better to make it clearer that there is one that should work normally...

**Review by mzanetti — CHANGES_REQUESTED** [review]
One issue spottet (I believe) and 2 tiny suggestions

File: aqi/integrationpluginaqi.cpp (hunk 1 above)
Comment (Boernsman) [reply aqi/integrationpluginaqi.cpp:43]: No, since there is a check if m_aqiConnection is not a nullptr

Comment (mzanetti) [reply aqi/integrationpluginaqi.cpp:43]: Are you really sure about that?

How I read this code it seems to be a nullptr when neither setupThing() nor discoverThings() has been called, which might be the case on a fresh startup where no AQI things are configured.

Comment (mzanetti) [reply aqi/integrationpluginaqi.cpp:44]: stupid me... I missed the check for the valid aqiConnectiono in the if...

**Review by mzanetti — APPROVED** [review]
happroved

**Pull request merged**